### PR TITLE
kops e2e: stop hard-coding c4.large

### PR DESCRIPTION
**Pull request by justinsb**
Otherwise we're not testing the kops behaviour: kops can auto-detect
whether instance types should be available in certain regions, and
chooses between m3 and c4 instances for the master.

Note that this isn't (currently) stock-out detection, but stock-outs are
a problem in kops as well so we want to get them under testing so we can
fix them.

### CODE REVIEW

**Comment by krzyzacy** [comment]
how does it work with other jobs does not have a `--kops-master-size` flag? they still need to be set to c4.large?

File: kubetest/kops.go [hunk 1]
```diff
@@ -64,7 +61,7 @@ var (
 	kopsVersion      = flag.String("kops-version", "", "(kops only) URL to a file containing a valid kops-base-url")
 	kopsDiskSize     = flag.Int("kops-disk-size", 48, "(kops only) Disk size to use for nodes and masters")
 	kopsPublish      = flag.String("kops-publish", "", "(kops only) Publish kops version to the specified gs:// path on success")
-	kopsMasterSize   = flag.String("kops-master-size", kopsAWSMasterSize, "(kops only) master instance type")
+	kopsMasterSize   = flag.String("kops-master-size", "", "(kops only) master instance type")
```
Comment (oomichi) [inline kubetest/kops.go]: How about adding

```
If not specifying, kops chooses instance type automatically for master
```

**Comment by justinsb** [comment]
So kops has some logic to auto-choose an instance type.  It won't use m3 instances in regions that don't support m3 instances.  (This is different from stock-outs, where there's no API - this is when regions just don't have the older instance types).

For a job that wants to specify the instance type, yes it needs to specify the kops-master-size.  Hence the c5 job, which is actually checking the c5 instances, does need to specify c5.large.

But the default should be to test what end-users hit with kops, in my opinion.

**Comment by fejta** [comment]
/lint

Is this PR still alive?

**Comment by BenTheElder** [comment]
[verifying max_connections fix]
/test pull-test-infra-bazel-canary
/test pull-test-infra-lint

Also.. ping @justinsb seems like we should finish this PR 🙃

**Pull request closed without merging**